src/frontend/planner_test/tests/testdata/subquery_expr_correlated.yaml

```diff
@@ -461,11 +461,11 @@
           └─LogicalAgg { group_key: [a.a3, b.b2], aggs: [count(1:Int32)] }
             └─LogicalJoin { type: LeftOuter, on: IsNotDistinctFrom(a.a3, c.c3) AND IsNotDistinctFrom(b.b2, c.c2), output: [a.a3, b.b2, 1:Int32] }
               ├─LogicalAgg { group_key: [a.a3, b.b2], aggs: [] }
-              | └─LogicalJoin { type: Inner, on: true, output: all }
+              | └─LogicalJoin { type: Inner, on: (a.a3 = b.b2), output: all }
```

Comment:
Not supposed to happen. Lets investigate why (might be to do with subquery).

Reply:
Actually, this is correct. The filter was actually derived from the inner join above. If you stare at the plan long enough you will see that the filter derivation and pushdown is indeed correct:
- we want to filter out those rows of `a, b` such that a3 = b2, and that table c has 3 elements such that `c2 = b2 = a3 = c3`